wolfcrypt/src/sha256.c

```diff
@@ -1732,7 +1740,11 @@ static int InitSha256(wc_Sha256* sha256)
     #endif
         XMEMCPY(hash, sha256->digest, WC_SHA256_DIGEST_SIZE);
 
+    #ifdef WOLF_CRYPTO_CB /* Use to reset state but keep devId info */
```

Comment:
Consider fixing `InitSha256` to not wipe out devId... and move the `sha256->devId = wc_CryptoCb_DefaultDevID();` login into `wc_InitSha256_ex`.

Reply:
I think it can be removed too. This was originally added to handle the cases where Sha256 didn't have a devId like CalcHashId. However those one-shot hash API's like `wc_Sha256Hash` now have this logic built-in. `devId = wc_CryptoCb_DefaultDevID();`